EIPS/eip-5089.md

```diff
@@ -0,0 +1,393 @@
+
+MUST NOT account for withdrawal limits like those returned from maxWithdraw and should always act as though the withdrawal would be accepted, regardless if the user has enough principal tokens, etc.
+
+MUST be inclusive of withdrawal fees. Integrators should be aware of the existence of withdrawal fees.
+
+MUST NOT revert due to principal token contract specific user/global limits. MAY revert due to other conditions that would also cause `withdraw` to revert.
+
+Note that any unfavorable discrepancy between `convertToPrincipal` and `previewWithdraw` SHOULD be considered slippage in price-per-principal-token or some other type of condition.
+
+```yaml
+- name: previewWithdraw
+  type: function
+  stateMutability: view
+
+  inputs:
+    - name: underlyingAmount
+      type: uint256
+
+  outputs:
+    - name: principalAmount
+      type: uint256
+```
+
+#### `withdraw`
+
+Burns `principalAmount` from `owner` and sends exactly `underlyingAmount` of underlying tokens to `receiver`.
+
+MUST emit the `Redeem` event.
+
+MUST support a withdraw flow where the principal tokens are burned from `owner` directly where `owner` is `msg.sender` or `msg.sender` has [EIP-20](./eip-20.md) approval over the principal tokens of `owner`.
+ MAY support an additional flow in which the principal tokens are transferred to the principal token contract before the `withdraw` execution, and are accounted for during `withdraw`.
+
+MUST revert if all of `underlyingAmount` cannot be withdrawn (due to withdrawal limit being reached, slippage, the owner not having enough principal tokens, etc).
+
+Note that some implementations will require pre-requesting to the principal token contract before a withdrawal may be performed. Those methods should be performed separately.
+
+```yaml
+- name: withdraw
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: underlyingAmount
+      type: uint256
+    - name: receiver
+      type: address
+    - name: owner
+      type: address
+
+  outputs:
+    - name: principalAmount
+      type: uint256
+```
+
+### Events
+
+#### Redeem
+
+`from` has exchanged `principalAmount` of Principal Tokens for `underlyingAmount` of underlying, and transferred that underlying to `to`.
+
+MUST be emitted when Principal Tokens are burnt and underlying is withdrawn from the contract in the `EIP5089.redeem` method.
+
+```yaml
+- name: Redeem
+  type: event
+
+  inputs:
+    - name: from
+      indexed: true
+      type: address
+    - name: to
+      indexed: true
+      type: address
+    - name: amount
+      indexed: false
+      type: uint256
+```
+
+#### Mature
+The Principal Token has hit maturity, and a a state change has been recorded on the Principal Token contract for the first time post-maturity.
```

Comment:
How would we know it's the first time/

Reply:
Uggh, this likely means that the implementation will have to add a storage variable and check it every time with every call to all transactional external fns which would be 2100 gas for a cold SLOAD.  

Additionally, the event doesn't get emitted exactly at maturity, it is emitted at an arbitrary time after maturity (whenever someone calls a transactional external fn).  

Furthermore, from the front end perspective, if we really wanted to know that arbitrary first transaction after maturity, it is trivial to identify it.

Reply:
@alcueca ^^  tagging you for visibility because this was marked as resolved

Reply:
Mmm, you are right, for analytics just retrieving the oldest `Redeem` event would work just fine. Also the actual maturing of the series is a completely predictable event, for which `Mature` is just a lagging indicator.